pkg/machine/config.go

```diff
@@ -137,6 +138,29 @@ type VM interface {
 	Stop(name string, opts StopOptions) error
 }
 
+func SetLock(lf **lockfile.LockFile, name string, vmtype VMType) error {
```

Comment:
Please add a comment/doc.  Why `**`?

Reply:
This seems like it would be much easier written as
`GetLock(name string, vmtype VMType) (*lockfile.LockFile,error)`
and the caller assigns the returned lock to the lock struct field.

Reply:
Concur, dropping the double-pointer seems a lot simpler